openfisca_france/model/prelevements_obligatoires/impot_revenu/ir.py

```diff
@@ -3211,6 +3211,83 @@ class nbptr(Variable):
     reference = 'http://vosdroits.service-public.fr/particuliers/F2705.xhtml'
     definition_period = YEAR
 
+    def formula_2008_01_01(foyer_fiscal, period, parameters):
+        nb_pac = foyer_fiscal('nb_pac', period)
```

Comment:
Pour les très longues formules comme ça, est-ce que ça ajouterait à la lisibilité du code de rajouter un petit commentaire expliquant la diff avec la formule d'avant ?
```suggestion
        '''
        Modification de la prise en compte des enfants à charge du conjoint décédé et du veuf.
        '''
        nb_pac = foyer_fiscal('nb_pac', period)
```

Reply:
Je me demande même si cela ne vaut pas le coup de factoriser le code.

Reply:
Je ne comprends pas ce que signifie factoriser le code

Reply:
Eviter d'avoir une grosse partie qui est copié-collé et donc ne l'avoir qu'à un seul endroit.
Cela suppose peut-être d'introduire une variable intermédiaire (qu'il faudra mettre dans la blacklist pour ne pas la conserver) ou une fonction ad hoc qu'on utilise au deux endroits (mais cela limite un peu la lisibilité du code pour des personnes non-spécialiste).

Reply:
je comprends l'idée mais vous ne pensez pas que factoriser réduirait la transparence de la formule ? au final, si on suit la brochure pratique de l'IR, le calcul est long...

Reply:
Moi je suis pour garder le changement de formule tel quel mais rajouter des explications comme le suggérait @pzuldp. Surtout que bon il n'y a pas des changements tous les ans non plus... Ça t'irait @benjello ?

Reply:
Tout me va.